### DISCUSSION TITLE
otlptracehttp modifying endpoint

### DISCUSSION THREAD

**Discussion by julianocosta89** [open-telemetry/opentelemetry-go#2597] [General]
Hello all! 👋🏽 

I'd like to open a discussion about something that I recently came across.
Is there a reason behind the options of otlptracehttp have the function [WithURLPath](https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlptrace/otlptracehttp/options.go#L71)?

Also, any reason the scheme being added to the endpoint?

I mean, if the user define the endpoint to something like: `https://otel-collector:4317` the config is failing, because it automatically prepends `https://`, resulting in the URL `https://https/://otel-collector:4317`.

As of the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#configuration-options), I don't believe it should work like that, am I wrong?

For Endpoint (OTLP/HTTP), we have the following:

> Target URL to which the exporter is going to send spans or metrics. The endpoint MUST be a valid URL with scheme (http or https) and host, MAY contain a port, SHOULD contain a path and MUST NOT contain other parts (such as query string or fragment). A scheme of https indicates a secure connection. When using OTEL_EXPORTER_OTLP_ENDPOINT, exporters MUST construct per-signal URLs as [described below](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#per-signal-urls). The per-signal endpoint configuration options take precedence and can be used to override this behavior (the URL is used as-is for them, without any modifications). See the [OTLP Specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md#otlphttp-request) for more details.

**The endpoint MUST be a valid URL with scheme (http or https) and host, MAY contain a port, SHOULD contain a path and MUST NOT contain other parts (such as query string or fragment)**.
I believe when we modify the endpoint adding or removing the scheme, we are going against that paragraph.

Can't we simply use the `endpoint` as it is when using the [`WithEndpoint`](https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlptrace/otlptracehttp/options.go#L60) function?

**Comment by busgo** [comment]
The endpoint  is http(s)://host+port    
url path is  path  not full url    default is  "/v1/trace"   "/v1/mertrics"

**Reply by julianocosta89** [reply]
If you add http(s) to the endpoint, it will fail.
The go implementation adds the protocol automatically.
That's exactly the first point of my question.
We shouldn't modify the endpoint.
